### PR TITLE
Fix issue with duplicate network paths in vSphere

### PR DESCRIPTION
**Pull request by saimanoj01**
### What this PR does / why we need it
Fix issue with duplicate network paths in vSphere 

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/1488

### Describe testing done for PR
Tested the below scenarios:
### Scenario 1
Network setup on vSphere:
image

TKG backend API response:
image

### Scenario 2
Network setup on vSphere:
image

TKG backend API response:
image

### Scenario 3
Network setup on vSphere:
image

TKG backend API response:
image

### Scenario 4
Network setup on vSphere:
image

TKG backend API response:
image

### Release note

```release-note
Fix issue with duplicate network paths in vSphere 
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### Additional information
There is a followup issue opened for adding/tracking unit tests - https://github.com/vmware-tanzu/tanzu-framework/issues/926

### CODE REVIEW

File: pkg/v1/tkg/vc/client.go [hunk 1]
```diff
@@ -460,14 +448,79 @@ func (c *DefaultClient) GetNetworks(ctx context.Context, datacenterMOID string)
...
+				name, err := c.getNetworkNameWithVirtualSwitch(ctx, results[i].Name, ref, commonProps)
+				if err != nil {
+					changeNetworkNameToMoid(results[i])
+					continue
+				}
+
+				// check if duplicate networks exist with the same path
+				_, err = c.FindNetwork(ctx, name, "")
+				if _, ok := err.(*find.MultipleFoundError); ok {
+					// if duplicate networks with the same path are found
```
Comment (saimanoj01) [inline pkg/v1/tkg/vc/client.go:468]: Expand more on this

**Review by prkalle — APPROVED** [review]
LGTM.
As discussed please add some comments to explain the logic in determining the duplicate networks. Thanks!

**Review by anujc25 — APPROVED** [review]

**Comment by heww** [comment]
Do you test this PR?  

Do we need to change the moid to type:id format?  Because of the path for the finder.Network only accepts the following format.

```
// Network finds a NetworkReference using a Name, Inventory Path, ManagedObject ID, Logical Switch UUID or Segment ID.
// With standard vSphere networking, Portgroups cannot have the same name within the same network folder.
// With NSX, Portgroups can have the same name, even within the same Switch. In this case, using an inventory path
// results in a MultipleFoundError. A MOID, switch UUID or segment ID can be used instead, as both are unique.
// See also: https://kb.vmware.com/s/article/79872#Duplicate_names
// Examples:
// - Name:                "dvpg-1"
// - Inventory Path:      "vds-1/dvpg-1"
// - ManagedObject ID:    "DistributedVirtualPortgroup:dvportgroup-53"
// - Logical Switch UUID: "da2a59b8-2450-4cb2-b5cc-79c4c1d2144c"
// - Segment ID:          "/infra/segments/vnet_ce50e69b-1784-4a14-9206-ffd7f1f146f7"
func (f *Finder) Network(ctx context.Context, path string) (object.NetworkReference, error) {
	networks, err := f.NetworkList(ctx, path)
	if err != nil {
		if _, ok := err.(*NotFoundError); ok {
			net, nerr := f.networkByID(ctx, path)
			if nerr == nil {
				return net, nil
			}
		}
		return nil, err
	}

	if len(networks) > 1 {
		return nil, &MultipleFoundError{"network", path}
	}

	return networks[0], nil
}
```

If the moid of the duplicated network is not format with type:id, the creation of the cluster will fail with the following message.
```
vSphere config validation failed: vSphere resources validation failed: invalid VSPHERE_NETWORK: network 'network-43' not found
```


This PR uses the type:id as the format of the moid https://github.com/vmware-tanzu/tanzu-framework/pull/1142

**Comment by heww** [comment]
@saimanoj01 I also find that there is a deficiency in PR https://github.com/vmware-tanzu/tanzu-framework/pull/1349 which will convert the managed object id (type:id) network to the path. For example, it will convert **Network:network-31** to **/kubo-dc/network/VM Network** and cause the cluster creation to fail.

**Pull request merged**